packages/client-sdk-nodejs/src/config/transport/grpc-configuration.ts

```diff
@@ -9,6 +9,12 @@ export interface GrpcConfigurationProps {
    * more than this amount will return a ResourceExhausted error.
    */
   maxSessionMemoryMb: number;
+
+  /**
+   * The number of internal clients a cache client will create to communicate with Momento. More of them allows
+   * more concurrent requests, at the cost of more open connections and the latency of setting up each client.
+   */
+  numClients: number;
```

Comment:
This is technically a breaking change to public API. If we wanted to make this backward-compatible I think we would want to make it an optional field with a default value of 6.

I very much doubt that anyone is constructing any of these from scratch today so I think it's a safe change and probably worth it to keep things simple - but we really need to all start being very diligent about assessing changes like this for backward compat, and if you file a PR that is technically breaking backward compat, you need to clearly call it out in the PR and explain why you think it's the right thing to do so that others can weigh in.

Reply:
Good point. I wasn't thinking about people constructing the configuration from scratch.

Reply:
We should probably make it a default value. That will hide it better anyway. Technically it's still a breaking change since the interface has a new field, but I would question why anyone would make their own implementation of that.